src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/LocalFunctionTests.cs

```diff
@@ -299,5 +299,207 @@ int G()
                 Assert.Equal("error CS0103: The name 'value' does not exist in the current context", error);
             });
         }
+
+        [Fact]
+        [WorkItem(51056, "https://github.com/dotnet/roslyn/issues/51056")]
+        public void LocalFunctionWithParameterHidingOuterScopeVariable()
+        {
+            var source = @"
+using System;
+
+class C
+{
+    static void Main(string[] args)
+    {
+        new C().M(42);
+    }
+
+    void M(int x)
+    {
+        local(string.Empty);
+        local2();
+
+        void local(string x)
+        {
+            GetValue(out var something); // captures 'this' so causes use of a display class from outer scope
+            Console.WriteLine(x); // set breakpoint here and inspect `x` (should display a `string`, not an `int`)
+        }
+
+        void local2()
+        {
+            Console.Write(x);
+        }
+    }
+
+    void GetValue(out int x)
+    {
+        x = 42;
+    }
+}";
+            var compilation0 = CreateCompilation(source, options: TestOptions.DebugDll);
+
+            WithRuntimeInstance(compilation0, runtime =>
```

Comment:
>WithRuntimeInstance [](start = 12, length = 19)

It doesn't look like we are checking for expected type (string) anywhere. What is used to verify that we are getting right things?

Reply:
Made the signatures of the frames visible.

---
In reply to: [574682637](https://github.com/dotnet/roslyn/pull/51129#discussion_r574682637) [](ancestors = 574682637)